crates/iceberg-catalog/src/service/task_queue/mod.rs

```diff
@@ -1,132 +1,219 @@
 pub mod tabular_purge_queue;
 
 #[derive(Debug, Clone)]
+pub struct QueueConfig {
+    pub config: TaskQueueConfig,
+    pub channel_size: usize,
+}
+
+pub type TaskQueueProducer = Arc<
+    dyn Fn(async_channel::Receiver<Task>) -> BoxFuture<'static, crate::api::Result<()>>
+        + Send
+        + Sync
+        + 'static,
+>;
+
+pub const DEFAULT_CHANNEL_SIZE: usize = 1000;
+
+#[derive(Clone)]
+struct RegisteredQueue {
+    pub config: QueueConfig,
+    pub queue_task: TaskQueueProducer,
+}
+
+impl Debug for RegisteredQueue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RegisteredQueue")
+            .field("config", &self.config)
+            .field("queue_task", &"Fn(...)")
+            .finish()
+    }
+}
+
+#[derive(Clone, Debug)]
 pub struct TaskQueues {
-    tabular_expiration: tabular_expiration_queue::ExpirationQueue,
-    tabular_purge: tabular_purge_queue::TabularPurgeQueue,
+    queues: HashMap<&'static str, async_channel::Sender<Task>>,
+    queue: Arc<dyn TaskQueue + Send + Sync + 'static>,
+    registered_queues: HashMap<&'static str, RegisteredQueue>,
 }
 
 impl TaskQueues {
     #[must_use]
-    pub fn new(
-        expiration: tabular_expiration_queue::ExpirationQueue,
-        purge: tabular_purge_queue::TabularPurgeQueue,
-    ) -> Self {
+    pub fn new(queue: Arc<dyn TaskQueue + Send + Sync + 'static>) -> Self {
         Self {
-            tabular_expiration: expiration,
-            tabular_purge: purge,
+            queues: HashMap::new(),
+            queue,
+            registered_queues: HashMap::new(),
         }
     }
 
-    #[tracing::instrument(skip(self))]
-    pub(crate) async fn queue_tabular_expiration(
-        &self,
-        task: TabularExpirationInput,
-    ) -> crate::api::Result<()> {
-        self.tabular_expiration.enqueue(task).await
+    pub fn register_queue(
+        &mut self,
+        queue_name: &'static str,
+        config: QueueConfig,
+        queue_task: TaskQueueProducer,
+    ) {
+        self.registered_queues
+            .insert(queue_name, RegisteredQueue { config, queue_task });
     }
 
-    #[tracing::instrument(skip(self))]
-    pub(crate) async fn cancel_tabular_expiration(
-        &self,
-        filter: TaskFilter,
-    ) -> crate::api::Result<()> {
-        self.tabular_expiration.cancel_pending_tasks(filter).await
+    async fn run(self, poll_interval: Duration) {
+        loop {
+            for (queue_name, queue_tx) in &self.queues {
+                match self.queue.pick_new_task(queue_name).await {
+                    Ok(Some(task)) => {
+                        if let Err(e) = tryhard::retry_fn(|| queue_tx.send(task.clone()))
+                            .retries(5)
+                            .fixed_backoff(Duration::from_millis(100))
+                            .await
+                        {
+                            // TODO: Do we even want to record_failure for this?
```

Comment:
I think we should just log an error.
The task might be important. We don't want to progress its attempts and remove it from the active tasks table if we haven't actually tried to run it due to internal problems.